apps/emqx/src/emqx_channel.erl

```diff
@@ -1011,6 +992,79 @@ not_nacked({deliver, _Topic, Msg}) ->
             true
     end.
 
+%%--------------------------------------------------------------------
+%% Handle Frame Error
+%%--------------------------------------------------------------------
+
+handle_frame_error(
+    Reason,
+    Channel = #channel{
+        conn_state = ConnState,
+        conninfo = ConnInfo
+    }
+) when
+    is_map(Reason) andalso
+        (ConnState == idle orelse ConnState == connecting)
+->
+    ShutdownCount = shutdown_count(frame_error, Reason),
+    case proto_ver(Reason, ConnInfo) of
+        ?MQTT_PROTO_V5 ->
+            shutdown(
+                ShutdownCount,
+                ?CONNACK_PACKET(connack_reason_code(Reason)),
+                Channel#channel{conninfo = ConnInfo#{proto_ver => ?MQTT_PROTO_V5}}
```

Comment:
nit: update channel with proto_ver for both this line and 1020 outside of the `case` block.